notebook/tests/selenium/utils.py

```diff
@@ -0,0 +1,163 @@
+import os
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+
+from contextlib import contextmanager
+
+pjoin = os.path.join
+
+
+def wait_for_selector(browser, selector, timeout=10, visible=False, single=False):
+    wait = WebDriverWait(browser, timeout)
+    if single:
+        if visible:
+            conditional = EC.visibility_of_element_located
+        else:
+            conditional = EC.presence_of_element_located
+    else:
+        if visible:
+            conditional = EC.visibility_of_all_elements_located
+        else:
+            conditional = EC.presence_of_all_elements_located
+    return wait.until(conditional((By.CSS_SELECTOR, selector)))
+
+
+class CellTypeError(ValueError):
+    
+    def __init__(self, message=""):
+        self.message = message
+
+class Notebook:
+    
+    def __init__(self, browser):
+        self.browser = browser
+        self.remove_safety_check()
+
+    @property
+    def body(self):
+        return self.browser.find_element_by_tag_name("body")
+
+    @property
+    def cells(self):
+        """Gets all cells once they are visible.
+        
+        """
+        wait_for_selector(self.browser, ".cell")
+        return self.browser.find_elements_by_class_name("cell")
+
+
+    @property
+    def current_cell_index(self):
+        return self.cells.index(self.current_cell)
+
+    def remove_safety_check(self):
```

Comment:
'safety' is rather vague. `remove_unsaved_check`, maybe? Or base it on the hook name: `remove_onbeforeunload`.